src/map/atcommand.cpp

```diff
@@ -7051,25 +7051,62 @@ ACMD_FUNC(autoloottype)
 	}
 
 	if (action < 3) { // add or remove
-		if ((strncmp(message, "healing", 3) == 0) || (atoi(message) == 0))
-			type = IT_HEALING;
-		else if ((strncmp(message, "usable", 3) == 0) || (atoi(message) == 2))
-			type = IT_USABLE;
-		else if ((strncmp(message, "etc", 3) == 0) || (atoi(message) == 3))
-			type = IT_ETC;
-		else if ((strncmp(message, "armor", 3) == 0) || (atoi(message) == 4))
+		char* ptr = nullptr;
+		int32 val = strtol( message, &ptr, 10 );
+
+		// If there was only a number in the input and no other characters
+		if( ptr != nullptr && *ptr == '\0' ){
```

Comment:
I found this part a bit hard to understand, because the variable "ptr" doesn't really make it clear that it's the last character from the conversion and it's not easy to understand why the value is a number when the last character is '\0'.

Also wouldn't this also be true if the value is empty string?

The c++ reference says that on fail: "the value of str is stored in the object pointed to by str_end."